attestationreport/attestationreport.go

```diff
@@ -654,6 +660,27 @@ func Verify(arRaw string, nonce, casPem []byte) VerificationResult {
 			result.Success = false
 		}
 	}
+
+	// Validate every specified policy
+	for _, policy := range policies {
+		pv, ok := policy.(PolicyValidator)
+		if !ok {
+			msg := "Internal Error: policy does not implement PolicyValidator"
+			log.Warn(msg)
+			result.Success = false
+			result.ProcessingError = append(result.ProcessingError, msg)
+			continue
+		}
+
+		err := pv.Validate(result)
+		if err != nil {
+			result.Success = false
+			// TODO should be two different types, result without success and policy validation
+			// which are merged together here
```

Comment:
I do not get the meaning of this comment. Do you mean that we plan to add a separate policyValidationResult as a second structure in parallel to what is now in the VerificationResult and then provide both results to the querying party?